dataset_loaders/images/mnistm.py

```diff
@@ -0,0 +1,183 @@
+"""Mnist-m dataset."""
+import numpy as np
+import os
+import time
+from PIL import Image
+from images.mnist import MnistDataset
+
+
+class MnistMDataset(MnistDataset):
+    """The mnist-m handwritten digit dataset.
+
+    The dataset should be downloaded from [1]
+
+    Parameters
+    ----------
+    which_set: string
+        A string in ['train', 'val', 'valid', 'test'], corresponding to
+        the set to be returned.
+
+     References
+    ----------
+    [1] Mnist-m dataset:
+    https://drive.google.com/file/d/0B9Z4d7lAwbnTNDdNeFlERWRGNVk/view
+    """
+
+    name = 'mnistm'
+
+    # optional arguments
+    data_shape = (28, 28, 3)
+
+    GTclasses = range(2)
+
+    non_void_nclasses = 2
+    _void_labels = []
+
+    _mask_labels = {
+        0: 'background',
+        1: 'digit',
+        }
+
+    _cmap = {
+        0: (0),           # background
+        1: (255),           # digit
+        }
+
+    _filenames = None
+
+    def __init__(self, which_set='train', *args, **kwargs):
```

Comment:
This method is the same as the one in `MnistDataset`. There is no need to redefine it. 
The same goes with all the other methods: remove the parts that do not change.